### PR TITLE
feat: Add sub config for TypeScript projects

### PR DESCRIPTION
**Pull request by susnux**
This adds a sub-config for typescript projects, e.g. we could use it within our libraries or for users which use TypeScript for app development (vue + ts is pretty common nowadays).

### CODE REVIEW

**Comment by skjnldsv** [comment]
Can't we just support typescript out of the box?
Surely there is a way to define different rules for ts and js files, no?

**Comment by susnux** [comment]
> Can't we just support typescript out of the box? Surely there is a way to define different rules for ts and js files, no?

@skjnldsv Good point! I changed it to simply use `overrides`.

File: package.json [hunk 1]
```diff
@@ -7,30 +7,38 @@
     "type": "git",
     "url": "git+https://github.com/nextcloud/eslint-config.git"
   },
+  "scripts": {
+    "lint": "eslint -c index.js *.js",
+    "lint:fix": "eslint -c index.js --fix *.js"
+  },
```
Comment (skjnldsv) [inline package.json:13]: Why is this here? It doesn't serve any purpose, no? :p

Comment (susnux) [reply package.json]: It does, but it does not belong to this pull request ;-)
Removed that changes from this PR

Comment (skjnldsv) [reply package.json]: Makes more sense! :grin:

**Comment by skjnldsv** [comment]
Could we maybe disable jsdoc for ts files?

**Comment by susnux** [comment]
> Could we maybe disable jsdoc for ts files?

This is the same behavior as for js file, why should we treat ts file differently regarding documentation?
One reason I see is the built in type information, thats why I disabled `jsdoc/require-param-type`.
(I am just curious)

**Comment by skjnldsv** [comment]
> For exported functions I think warning on missing parameter documentation is good,

Ah right, typing requirements is bad, but param documentation is still quite good to have! :+1:

**Review by skjnldsv — APPROVED** [review]

File: index.js [hunk 2]
```diff
@@ -116,4 +116,39 @@ module.exports = {
...
+			files: ['**/*.ts'],
+			extends: [
+				'@vue/eslint-config-typescript/recommended',
+				'plugin:import/typescript',
+			],
+			parserOptions: {
+				parser: '@typescript-eslint/parser',
+			},
+			rules: {
+				'n/no-missing-import': 'off',
```
Comment (skjnldsv) [inline index.js:130]: @susnux why disabling this rule btw?
If we cannot resolve, maybe it's bad?
